engine/classes/Elgg/Http/ResponseFactory.php

```diff
@@ -0,0 +1,410 @@
+<?php
+
+namespace Elgg\Http;
+
+use Elgg\Ajax\Service as AjaxService;
+use Elgg\PluginHooksService;
+use ElggEntity;
+use InvalidParameterException;
+use SecurityException;
+use Symfony\Component\HttpFoundation\AcceptHeader;
+use Symfony\Component\HttpFoundation\RedirectResponse as SymfonyRedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * @since 2.2.0
+ * @access private
+ */
+class ResponseFactory {
+
+	/**
+	 * @var Request
+	 */
+	private $request;
+
+	/**
+	 * @var AjaxService
+	 */
+	private $ajax;
+
+	/**
+	 * @var PluginHooksService
+	 */
+	private $hooks;
+
+	/**
+	 * @var bool 
+	 */
+	private $response_sent = false;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param Request            $request HTTP request
+	 * @param PluginHooksService $hooks   Plugin hooks service
+	 * @param AjaxService        $ajax    AJAX service
+	 */
+	public function __construct(Request $request, PluginHooksService $hooks, AjaxService $ajax) {
+		$this->request = $request;
+		$this->hooks = $hooks;
+		$this->ajax = $ajax;
+	}
+
+	/**
+	 * Creates an HTTP response
+	 *
+	 * @param string  $content The response content
+	 * @param integer $status  The response status code
+	 * @param array   $headers An array of response headers
+	 * @return Response
+	 */
+	public function prepareResponse($content = '', $status = 200, array $headers = array()) {
+		$response = new Response($content, $status, $headers);
+		$response->prepare($this->request);
+
+		// Read the accept header and set the Content-Type of the response accordingly
+		$accept = AcceptHeader::fromString($this->request->headers->get('Accept'))->all();
```

Comment:
This is just temporary until I rewire things to use JsonResponse instead. I think we might need to hold off sending the response and let calling code call send after setting appropriate headers